File: HerenciaCSharp/HerenciaCSharp/Profesor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HerenciaCSharp
{
    public class Profesor: Persona  
    {
        private String Especialidad;

        public void setEspecialidad(String val)
        {
            Especialidad = val;
        }

        public String getEspecialidad()
        {
            return Especialidad;
        }
    }
}

File: HerenciaVisual/HerenciaVisual/Profesor.vb
Public Class Profesor
    Inherits Persona

    Private especialidad As String

    Sub New()
    End Sub

    Sub New(ByVal especialidad1 As String)
        Me.especialidad = especialidad1
    End Sub

    Public Sub SetEspecialidad(ByVal especialidad1 As String)
        especialidad = especialidad1
    End Sub

    Public Function GetEspecialidad() As String
        Return especialidad
    End Function
End Class
